src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -130,14 +152,26 @@ private static FlutterLogEntry parseGCEvent(@NotNull Event event) {
     final long timeMs = Math.round(time * 1000);
     final double usedMB = used / (1024.0 * 1024.0);
     final double maxMB = maxHeap / (1024.0 * 1024.0);
+    final String message = "collection time " + nf.format(timeMs) + "ms • " +  df1.format(usedMB) + "MB used of " + df1.format(maxMB) + "MB • " + isolateRef.getId();
 
-    return new FlutterLogEntry(
-      timestamp(event),
-      "runtime.gc", GC_EVENT_LEVEL,
-      "collection time " +
-      nf.format(timeMs) + "ms • " +
-      df1.format(usedMB) + "MB used of " + df1.format(maxMB) + "MB • " +
-      isolateRef.getId());
+    final LineInfo lineInfo = parseLine(message, GC_CATEGORY);
+    return new FlutterLogEntry(timestamp(), lineInfo, UNDEFINED_LEVEL.value);
+  }
+
+  private static Kind parseKind(@NotNull String message, @NotNull String category) {
+    if (category.equals(TOOLS_CATEGORY)) {
+      message = message.trim();
+      if (message.equals("Performing hot reload...") || message.equals("Initializing hot reload...")) {
+        return Kind.RELOAD;
+      }
+      if (message.equals("Performing hot restart...") || message.equals("Initializing hot restart...")) {
+        return Kind.RESTART;
+      }
+      if (message.startsWith("══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY")) {
```

Comment:
in terms of technical debt for the quarter can you file a bug to remove this string matching and reference the bug with a TODO here.

Reply:
Totally, will do.